src/OrchardCore.Modules/OrchardCore.Lists/Services/ContainerService.cs

```diff
@@ -339,9 +354,43 @@ public async Task<IEnumerable<ContentItem>> QueryContainedItemsAsync(string cont
             }
         }
 
-        private static Expression<Func<ContentItemIndex, bool>> CreateOrderedContentIndexFilter(bool publishedOnly)
+        private IQuery<ContentItem> ContainedItemOptionsFilter(ContainedItemOptions containedItemOptions, IQuery<ContentItem> containedItems)
+        {
+            if (!string.IsNullOrEmpty(containedItemOptions.DisplayText))
+            {
+                containedItems.With<ContentItemIndex>().Where(i => i.DisplayText.Contains(containedItemOptions.DisplayText));
+            }
+
+            if (containedItemOptions.Status != ContentsStatus.None)
+            {
+                switch (containedItemOptions.Status)
+                {
+                    case ContentsStatus.Draft:
+                        containedItems?.With<ContentItemIndex>().Where(i => !i.Published);
```

Comment:
query.With<ContentItemIndex>(x => x.Latest && !x.Published);

Otherwise you're also getting previous versions.

Also the `.Where` is unecesary here

Reply:
Alright.

Reply:
Alright.